tests/ciadmin/test_generate_worker_pools.py

```diff
@@ -212,6 +218,17 @@ def assert_guest_accelerators(pool):
             id="guest_accelerators",
         ),
         pytest.param("azure", None, id="azure_basic"),
+        pytest.param(
+            "azure",
+            {
+                "storageProfile": {
+                    "imageReference": {
+                        "id": "/subscriptions/108d46d5-fe9b-4850-9a7d-8c914aa6c1f0/rgroup_id/rg-packer-worker-images/providers/Microsoft.Compute/galleries/name_id/images/name_id/versions/ver_id"
```

Comment:
I suspect you don't want to put this config here. This `pytest.param` is the inputs to the pool generation, so I *think* this is the output you want to verify gets generated correct?

Assuming that's the case, you'll want to assert the result is equivalent to  this in your assertion function (see prior comment).